### DISCUSSION TITLE
How can I switch theme dynamically in my app.

### DISCUSSION THREAD

**Discussion by FSMargoo** [AvaloniaUI/Avalonia#14491] [Q&A] [answered]
I has problem in changing theme dynamically in avalonia ui (like dark&light mode switch), I can't find any documentation or example code about it(I found the samples but I can't build it on my PC). What API should I use to achieve my intended functionality.

**Comment by thevortexcloud** [comment, marked as answer]
Did you check the control catalogue? From memory it already does this.

**Reply by FSMargoo** [reply]
Yes, but this demo can't run on my PC... I can't sure that the sample's code out of date or not.

**Reply by thevortexcloud** [reply]
You need to clone the entire repo and set up all the sub modules for it to work. Follow the Linux instructions  if you are only testing on your PC (even if you are on WIndows), you don't need the rest of it.

https://github.com/AvaloniaUI/Avalonia/blob/master/Documentation/build.md#linuxmacos

**Reply by FSMargoo** [reply]
Oh, I found that I just downloaded the previous sample code, and the latest sample code can run, thank you :)

**Reply by thevortexcloud** [reply]
It's possible it's broken now then. Master is not always stable. You can use the latest release version by cloning the branch for it. EG https://github.com/AvaloniaUI/Avalonia/tree/release/11.0.7
